changelog.d/12570.bugfix

```diff
@@ -0,0 +1 @@
+Fix a bug introduced in Synapse 1.57 which could cause `Failed to calculate hosts in room` errors to be logged for outbound federation.
```

Comment:
Is the bug just that we did something silly and logged about it? E.g. did we stop sending any in-band events if we logged this line?

Reply:
https://github.com/matrix-org/synapse/issues/12509#issuecomment-1111202236 Seems to suggest there was no problem other than the logging.

Reply:
correct, the only problem is log spam.